backend/baseclass.pm

```diff
@@ -889,20 +892,23 @@ sub wait_serial {
     }
 
     $regexp = [$regexp] if ref $regexp ne 'ARRAY';
-    my $initial_time = time;
-    while (time < $initial_time + $timeout) {
+    my $initial_time  = time;
+    my $bytes_to_keep = 0;
```

Comment:
It's possible that the file grows during this function, which would then result in a wrong offset.

I think it's simpler and more reliable to save the current offset here and add the match end to that instead, i.e.:

```
my $current_offset = $self->{serial_offset};
...
            if (!$args->{no_regex} && $str =~ m/$r/) {
                $current_offset += $LAST_MATCH_END[0];
                last MATCH;
            }
...
$self->{serial_offset} = $current_offset;
```

Reply:
I have initially thought that `$self->{serial_offset}` should be handled within `set_serial_offset` as the code calls the method at the end anyway.

Reply:
I'd be in favor of dropping that method, but it's also used by `_handle_command_set_current_test`...

Maybe it could be renamed "clear_serial_buffer` or something like that.